src/rgw/rgw_sal_motr.cc

```diff
@@ -457,6 +457,49 @@ int MotrUser::store_user(const DoutPrefixProvider* dpp,
 
 int MotrUser::remove_user(const DoutPrefixProvider* dpp, optional_yield y)
 {
+  // Remove user info from cache
+  // Delete access keys for user
+  // Delete user info index
+  // Delete user index
+  // Delete email for user - TODO
+  bufferlist bl;
+  // Remove the user info from cache.
+  store->get_user_cache()->remove(dpp, info.user_id.id);
+
+  // Delete all access key of user
+  if (!info.access_keys.empty()) {
+    for(auto acc_key = info.access_keys.begin(); acc_key != info.access_keys.end(); acc_key++) {
+      auto access_key = acc_key->first;
+      int access_del_rc = store->do_idx_op_by_name(RGW_IAM_MOTR_ACCESS_KEY,
+                              M0_IC_DEL, access_key, bl);
+      if (access_del_rc < 0){
+        ldpp_dout(dpp, 0) << "Unable to delete access key" << access_del_rc << dendl;
+        return access_del_rc;
+      }
+    }
+  }
+  
+  // Delete user info index
+  string user_info_iname = "motr.rgw.user.info." + info.user_id.id;
+  store->delete_motr_idx_by_name(user_info_iname);
+  ldpp_dout(dpp, 10) << "Deleted user info index" << dendl;
```

Comment:
print user_info_iname in logs this will help to debug

Reply:
Added **user_info_iname** in log